### ISSUE TITLE
zfs bced7e3aaa3cf54d5e8e4f94e067144b27cb744b causes null pointer deref in spl_kmem_cache_create

### ISSUE THREAD

**Issue by prometheanfire** [zfsonlinux/zfs#9302]
# System information
Type | Version/Name
 --- | --- 
Distribution Name	| Gentoo
Distribution Version	| :smile: 
Linux Kernel	| 5.2.13
Architecture	| x86_64
ZFS Version	| b63e2d881f859f0c7d8596be2759dd096e324f48
SPL Version	| b63e2d881f859f0c7d8596be2759dd096e324f48

### Describe the problem you're observing
Cannot Boot with builtin kernel (I CAN build it though, unlike in #9297)
Had to boot with earlycon=efifb and console=efifb to see it (and took 15 minutes to read out at 1 line every couple seconds)

### Describe how to reproduce the problem
Build a 'built-in' kernel (static) with the commit refrenced above and try to boot.

### Include any warning/errors/backtraces from the system logs
Best I have is a picture.

https://photos.app.goo.gl/eAaV8fXuXXfcUqDE7

**Comment by prometheanfire** [comment]
a18f8bce5c6acb9c3990b2917efa96d1828e541c boots

**Comment by prometheanfire** [comment]
a57c82fc50bced77b7f409595130091360533012 boots (all those typo patches)

**Comment by prometheanfire** [comment]
d02186ee2b0570069cfbb0d08d5c52752648f388 boots (sept 04)

**Comment by prometheanfire** [comment]
870e7a52c105f26ef4254b90230d396f4ce39ea7 boots that leaves 4 commits

not 4bbf0477a7f8936627c37ab89c704e763aa63b37 as that's rpm only
not 43f4495bde1f862a7f234919fb6c627abfdbd040 as it's test only
not b63e2d881f859f0c7d8596be2759dd096e324f48 as it's test only

I'm not sure how, but it looks like bced7e3aaa3cf54d5e8e4f94e067144b27cb744b introduced it, testing now

**Comment by prometheanfire** [comment]
bced7e3aaa3cf54d5e8e4f94e067144b27cb744b worked but master did not, odd... the diff confirms only tests were touched...

**Comment by prometheanfire** [comment]
bced7e3aaa3cf54d5e8e4f94e067144b27cb744b DID cause the issue on recompile

**Comment by behlendorf** [comment]
Thanks for bisecting this!  It looks like bced7e3 accidentally reintroduced issue #7595, which was previously addressed by 517d2471921fc44622dc737b09b655829b8c9c6f.  Would it be possible for you to verify that reapplying the original fix addresses the issue.

```diff
diff --git a/copy-builtin b/copy-builtin
index c108cfc6733..9a35b1845c3 100755
--- a/copy-builtin
+++ b/copy-builtin
@@ -12,9 +12,15 @@ usage()
 KERNEL_DIR="$(readlink --canonicalize-existing "$1")"
 
 MODULES=()
+
+# When integrated in to the kernel the spl module must appear first to
+# ensure that it is initialized before the other kmods which depend on it.
+MODULES+="spl"
+
 for MODULE_DIR in module/* module/os/linux/*
 do
        [ -d "$MODULE_DIR" ] || continue
+       [ "spl" = "${MODULE_DIR##*/}" ] && continue
        [ "os" = "${MODULE_DIR#*/}" ] && continue
        MODULES+=("${MODULE_DIR#*/}")
 done
```

**Comment by prometheanfire** [comment]
heh, thought this sounded familiar, I'll test tonight when I update to 5.2.14
